File: frontend/src/services/assetsDashboard.ts
import { api } from "@/services/api";

export type AssetDashboardCountValue = {
  count: number;
  value: number;
};

export type AssetDashboardStatusKpis = {
  total: AssetDashboardCountValue;
  in_use: AssetDashboardCountValue;
  available: AssetDashboardCountValue;
  maintenance: AssetDashboardCountValue;
  lost_or_discarded: AssetDashboardCountValue;
};

export type AssetDashboardPhysicalRow = {
  condition: string;
  label: string;
  count: number;
  value: number;
};

export type AssetDashboardGroupRow = {
  key: string;
  label: string;
  count: number;
  value: number;
};

export type AssetDashboardCostCenterRow = {
  key: string;
  label: string;
  asset_count: number;
  amount_total: number;
  average_value: number;
};

export type AssetDashboardAlertSummary = {
  count: number;
  amount_total: number;
  damaged_count: number | null;
};

export type AssetDashboardAlerts = {
  expired_inspections: AssetDashboardAlertSummary;
  expiring_inspections: AssetDashboardAlertSummary;
  without_holder: AssetDashboardAlertSummary;
  fair_condition: AssetDashboardAlertSummary;
};

export type AssetDashboardRead = {
  status: AssetDashboardStatusKpis;
  physical_condition: AssetDashboardPhysicalRow[];
  by_category: AssetDashboardGroupRow[];
  by_cost_center: AssetDashboardCostCenterRow[];
  alerts: AssetDashboardAlerts;
};

export async function fetchAssetsDashboard(): Promise<AssetDashboardRead> {
  const { data } = await api.get<AssetDashboardRead>("/assets/dashboard");
  return data;
}

File: app/schemas/assets_dashboard.py
from __future__ import annotations

from pydantic import BaseModel


class AssetDashboardCountValue(BaseModel):
    count: int = 0
    value: float = 0.0


class AssetDashboardStatusKpis(BaseModel):
    total: AssetDashboardCountValue
    in_use: AssetDashboardCountValue
    available: AssetDashboardCountValue
    maintenance: AssetDashboardCountValue
    lost_or_discarded: AssetDashboardCountValue


class AssetDashboardPhysicalRow(BaseModel):
    condition: str
    label: str
    count: int = 0
    value: float = 0.0


class AssetDashboardGroupRow(BaseModel):
    key: str
    label: str
    count: int = 0
    value: float = 0.0


class AssetDashboardCostCenterRow(BaseModel):
    key: str
    label: str
    asset_count: int = 0
    amount_total: float = 0.0
    average_value: float = 0.0


class AssetDashboardAlertSummary(BaseModel):
    count: int = 0
    amount_total: float = 0.0
    damaged_count: int | None = None


class AssetDashboardAlerts(BaseModel):
    expired_inspections: AssetDashboardAlertSummary
    expiring_inspections: AssetDashboardAlertSummary
    without_holder: AssetDashboardAlertSummary
    fair_condition: AssetDashboardAlertSummary


class AssetDashboardRead(BaseModel):
    status: AssetDashboardStatusKpis
    physical_condition: list[AssetDashboardPhysicalRow]
    by_category: list[AssetDashboardGroupRow]
    by_cost_center: list[AssetDashboardCostCenterRow]
    alerts: AssetDashboardAlerts
